### PR TITLE
Set full path to protoc inside _gRPC_PROTOBUF_PROTOC variable, so it …

### PR DESCRIPTION
**Pull request by kskalski**
…can work for both MODULE and PACKAGE type of protobuf provider.

### CODE REVIEW

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

**Comment by grpc-testing** [comment]
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

File: CMakeLists.txt [hunk 1]
```diff
@@ -339,7 +339,7 @@ function(protobuf_generate_grpc_cpp)
              "${_gRPC_PROTO_GENS_DIR}/${RELFIL_WE}_mock.grpc.pb.h"
              "${_gRPC_PROTO_GENS_DIR}/${RELFIL_WE}.pb.cc"
              "${_gRPC_PROTO_GENS_DIR}/${RELFIL_WE}.pb.h"
-      COMMAND $<TARGET_FILE:${_gRPC_PROTOBUF_PROTOC}>
+      COMMAND ${_gRPC_PROTOBUF_PROTOC}
```
Comment (jtattermusch) [inline CMakeLists.txt]: why remove the $<TARGET_FILE:  >?

File: CMakeLists.txt [hunk 2]
```diff
@@ -189,7 +189,7 @@ if("${gRPC_PROTOBUF_PROVIDER}" STREQUAL "module")
       set(_gRPC_PROTOBUF_PROTOC_LIBRARIES libprotoc)
     endif()
     if(TARGET protoc)
-      set(_gRPC_PROTOBUF_PROTOC protoc)
+      set(_gRPC_PROTOBUF_PROTOC ${PROTOBUF_ROOT_DIR}/protoc)
```
Comment (jtattermusch) [inline CMakeLists.txt]: Is that really where protoc will be located if I build protobuf as a submodule?

Comment (kskalski) [inline CMakeLists.txt]: This variable is set to:
    set(PROTOBUF_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third_party/protobuf)
and that directory gets protoc placed in it:
$ ls third_party/protobuf/protoc -l
-rwxr-xr-x 1 pi pi 9397460 Jul 20 18:53 third_party/protobuf/protoc

At least this is what happens when I build with this setting

File: CMakeLists.txt (hunk 1 above)
Comment (kskalski) [inline CMakeLists.txt]: When running with cmake . -DgRPC_PROTOBUF_PROVIDER=package -DgRPC_ZLIB_PROVIDER=package -DgRPC_SSL_PROVIDER=package
I get

CMake Error at CMakeLists.txt:336 (add_custom_command):
  Error evaluating generator expression:

    $<TARGET_FILE:/usr/bin/protoc>

When {_gRPC_PROTOBUF_PROTOC} point to absolute path then prefixing it with anything doesn't make sense. I'm not sure what was the original intention of using TARGET_FILE here, I guess it accidentally pointed to the same directory as PROTOBUF_ROOT_DIR

**Comment by jtattermusch** [comment]
Should be fixed by https://github.com/grpc/grpc/pull/12411

**Pull request closed without merging**